update c8run shutdown, spring recommendations

conceptualshark opened the pull request:
## Description

<!-- Provide an overview of what to expect in the PR. -->
<!-- Relate or link the associated epic or task. -->
<!-- Add `@camunda/tech-writers` as reviewer to pull in a tech writer, or add your embedded tech writer. -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [ ] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for **an upcoming minor release** and:
  - [ ] are in the `/docs` directory (version 8.8).
  - [ ] are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [ ] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [ ] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

conceptualshark commented:
I haven't touched the C8Run components/snippets in this PR - they specifically say to start C8Run with the Java commands (though don't call out that it's Java), so don't have the same limitation as the Spring guide.

I'm running through the Spring steps now to see how it works if you _do_ use Docker, put but this in as a quick fix if needed.

akeller commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -37,13 +37,9 @@ Before getting started, ensure you have the following in your local environment:
-
-If no version of Java is found, follow your chosen installation's instructions for setting `JAVA_HOME` before continuing.
-:::
+  - Ensure you are running Camunda 8 Run as a Java application. The Docker Compose distribution is not yet supported.
```
Since the "default" way to onboard is through C8Run, we can probably just keep the first sentence regardless of your test, but I am curious if the Docker Compose method introduces friction.

akeller reviewed (approved):
Nothing blocking here, but admittedly, I'm still not sure how to position these options. My internal recommendation was to make the Java application method the "default" way, but I don't know if or how we can mention the Docker Compose option in the guide. 

If it's not tested, I think it's fair to say it isn't supported.

conceptualshark commented on docs/self-managed/setup/deploy/local/c8run.md:
```diff
@@ -55,25 +55,28 @@ If Camunda 8 Run fails to start, run the [shutdown script](#shut-down-camunda-8-
 
 All Camunda 8 Run components can be accessed using the username/password combination `demo`/`demo`.
 
+:::note
+The URLs for the Docker Compose application can be found in the [Docker Compose](/self-managed/setup/deploy/local/docker-compose.md#access-components) documentation.
+:::
```
@akeller Opinions on making this a tabbed experience to list both, vs just linking to the Docker docs? If using tabs, I wasn't sure if "Java" as a tab header for the default experience would be clear enough to every user that that's what they needed.

akeller replied:
@conceptualshark let's just link for now. I'd like to get a more comprehensive understanding of the E2E journey for C8 Run, and specifically the onboarding experience. (See [here](https://camunda.slack.com/archives/C03UR0V2R2M/p1743781745857859) and [here](https://camunda.slack.com/archives/C089AUH6065/p1743771189872659))

conceptualshark commented:
I tested this locally and Docker Compose seems to work with the Spring guide, it just needed some notes that the Component URLs were different, so I've removed the warning that it's not recommended and updated those instead.

akeller commented on versioned_docs/version-8.7/guides/getting-started-java-spring.md:
```diff
@@ -96,7 +96,7 @@ To deploy your process, take the following steps:
 5. Click **Deploy**.
 
-When you open Operate at http://localhost:8080/operate/, you should now note the process deployed to your local Self-Managed setup.
+When you open Operate at http://localhost:8080/operate/ (or http://localhost:8081 using Docker Compose), you should now note the process deployed to your local Self-Managed setup.
```
I want to keep the focus on C8Run as a Java application for now, but the difference in ports highlights an issue with the experience... 

Maybe instead, we could qualify this as "Camunda 8 Run with Docker Compose". WDYT?

conceptualshark replied:
I've updated it to be specific - how is this version?

akeller replied:
Fab!
